### PR TITLE
modify setupdockerhost to create network object based on the bridge s…

### PR DESCRIPTION
**Pull request by zet809**
…pecified for 1.10 and higher

### CODE REVIEW

**Comment by zet809** [comment]
@daniceexi Modified, pls help to review.

File: xCAT/postscripts/setupdockerhost [hunk 1]
```diff
@@ -140,8 +106,68 @@ fi
...
     echo "Docker service starting failed"
     exit 1
 fi
+
+# TO check whether the docker server version is lower than or equal to 1.10
+#    If so, need to create docker network object based on the bridge specified
+#    Else, use the brdige as an option for docker daemon
+
+
+# For docker 1.8.x and higher
```
Comment (daniceexi) [inline xCAT/postscripts/setupdockerhost]: meaningless comment

File: xCAT/postscripts/setupdockerhost [hunk 2]
```diff
@@ -140,8 +106,68 @@ fi
...
+DOCKER_VERSION=`docker info 2>/dev/null | grep "Server Version:" | awk -F: '{print $2}'`
+if [ -z "$DOCKER_VERSION" ];then
+    # For docker 1.7.x and lower
```
Comment (daniceexi) [inline xCAT/postscripts/setupdockerhost]: meaningless comment

File: xCAT/postscripts/setupdockerhost [hunk 3]
```diff
@@ -140,8 +106,68 @@ fi
...
+    DOCKER_VERSION=`docker version | grep "Server version" | awk -F: '{print $2}'`
+    if [ -z "$DOCKER_VERSION" ];then
+        echo "Failed to get docker server version"
+        exit 1
+    fi
+fi
+MAX_VERSION=`echo $DOCKER_VERSION | awk -F. '{print $1}'`
```
Comment (daniceexi) [inline xCAT/postscripts/setupdockerhost]: It should use Major, Minor version

File: xCAT/postscripts/setupdockerhost [hunk 4]
```diff
@@ -140,8 +106,68 @@ fi
...
+fi
+
+FLAG=0
+if [  $MAX_VERSION -le 1 -a $MIN_VERSION -le 10 ]; then
+    FLAG=1
+fi
+
+# The docker newer than 1.10.x can accept NICNAME directly.
+# But we treat it as 1.10.x in case it is not supported immediately.
+FLAG=1
```
Comment (daniceexi) [inline xCAT/postscripts/setupdockerhost]: why did you set it in static here?

**Comment by daniceexi** [comment]
I have reviewed the change and agreed to merge. Thanks.

**Comment by whowutwut** [comment]
@daniceexi @zet809 I just noticed the comments that were made is regarding checking the version of docker installed.  This might be a good time to bring up pull request #374 again.  Instead of doing these parsing of each major and minor version, can we look into the suggestion to use the Sort::Versions instead?   I think we should use the libraries that are provided rather than doing the parsing like we have done...

**Comment by daniceexi** [comment]
Yes, I agree with that. This is a shell script, we could make a version comparison function in xcatlib so that all the postscripts can use it.

**Pull request merged**